interview-questions.md

```diff
@@ -7,44 +7,84 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-  Your answer:
+  Your answer: 
+You can fix this easily with a migration, which is a file that you generate when making changes to the shape of the database. First you would generate the migration file, edit the file to add the changes you want. In this case, you would add a column for Cohort_id and integer as the data type under the Student model, because the foreign id is used to associate the row of data in the Student model to a particular Cohort in the Cohort model through that Cohort's primary key.
+After you edit the file, run the migration command then confirm if the schema is updated with the new column under the Student model.
 
   Researched answer:
+You can make changes to the shape of the database, or model, with migrations.
+
+First you would generate the migration file in terminal:
+
+$ rails generate migration add_column_for_foreign_key
+
+after the word migration you would add a description of what change you are making in snake_case.
+after the file is generated, edit the file under db/migrate and add between the class tags (under the Student model)
+
+add_column :students, :cohort_id, :integer
+
+First key ":students" refers to the name of the table you're editing
+Tecond key ":cohort_id" is the name of the column you're adding. When you add a foreign key column, it needs to be the name of the table it's associated with (singular form) followed by _id. The foreign key goes to the students model because the foreign key is used to refer to the associated Cohort by the specific Cohort's primary key.
+Third key is the data type, which for foreign key you'd need to set as an integer
 
+Once you save the changes, go back to terminal and run the migration:
+rails db:migrate
+
+and then check your schema file to confirm that the new cohort_id column has been added to the Student model.
 
 
 2. Which RESTful routes must always be passed params? Why?
 
   Your answer:
+From what I remember working with full stack Rails apps, routes that need params, or a parameters to be passed through the url, would be used for specific data entries in the database. For example for the blog app prototype I worked on, a singular blog post that needs to be viewed, edited or deleted will use RESTful routes: GET, PATCH or DELETE which needs to use the id of the blog post. The ID can be passed to the url using params with the route including "blogs/:id" The HTTP verb CREATE doesn't need params since it doesn't need a particular blog post entry to be passed through the url. You're creating a new blog post of which the id doesn't exist yet.
 
   Researched answer:
-
+The RESTful routes that must always be passed params include GET, PATCH or DELETE HTTP verb requests.
```

Comment:
Those http verbs are true in particular situations. Please review the following site to see those situations. 
https://github.com/learn-academy-2022-alpha/Syllabus/blob/main/rails/restful-routes-crud.md